### ISSUE TITLE
Enhancement Localization tool

### ISSUE THREAD

**Issue by WouterStraver** [mfractor/mfractor-feedback#134]
Would it be possible to use CMD + D (Go to Declaration) on a translation binding in Xaml and when pressed navigate to the resource file and the correct line. Or another function to easily overwrite a translation.

**Comment by WouterStraver** [comment]
Also a preview on hover for the translation would be awesome as well ;-)

**Comment by matthewrdev** [comment]
This is a great suggestion @WouterStraver. I'll be doing a polish pass over MFractor soon and I'll aim to include both of these suggestions.

This may be a few weeks away as we are currently rolling out our Windows alpha release.

**Comment by matthewrdev** [comment]
@WouterStraver 

Following up on this, would you also like us to detect when you have a `TranslateExtension` defined and suggest localisation strings for you?

**Comment by WouterStraver** [comment]
That would be even better!

**Comment by WouterStraver** [comment]
Is it also possible to give a warning when missing one of the translations in one of the language files?

**Comment by matthewrdev** [comment]
Yes, we can do this as well.

**Comment by matthewrdev** [comment]
Localisation Tooling Enhancements:
 - [ ] When over a localisation expression, show localisation values in a tooltip.
 - [ ] Go to definition support for localisation expressions.
 - [ ] When a translate extension is available, detect resx keys and suggest in intellisense.
 - [ ] Code analyser to inspect the `Translate` extension usage and detect when localisation values are missing for locales for a given key.
 - [ ] Code fix to generate a resx entry (or entries) for missing/unknown localisation keys.

**Comment by matthewrdev** [comment]
@WouterStraver The recommended practices for localisation in Xamarin.Forms has changed since this issue was filed: https://docs.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/localization/text?pivots=macos

The `TranslateExtension` and `ILocale` methodology are no longer recommended. Instead, using an `x:Static` expression to locate a resource is the new recommended approach.

As a result, I will be revising the localisation tooling to follow these suggested practices.

**Comment by matthewrdev** [comment]
Example localisation expression:

```
<Label Text="{x:Static resources:AppResources.NotesLabel}" />
```

Where:
 * `AppResources` resolves to the code-behind component of a resx file.
 * `NotesLabel` is the key for a localisation lookup.

Localisation Tooling Enhancements:

 - [ ] When over a localisation expression, show localisation values summary in a tooltip.
 - [ ] Go to definition support for localisation expressions.
 - [ ] When localisation (RESX) values are available, suggest RESX values in code completion and inject the 
 - [ ] When localisation values are available, inspect string literals and match them to known localisation values.
 - [ ] Code action to move a string literal to a RESX file (new or existing)

**Comment by matthewrdev** [comment]
@WouterStraver Navigation and tooltip support is completed and will be available in MFractor 4.3.

<img width="890" alt="Screen Shot 2020-07-31 at 8 56 59 am" src="https://user-images.githubusercontent.com/3260473/88982531-df23fc80-d30b-11ea-86de-6fc6578aa98c.png">

**Comment by WouterStraver** [comment]
Very nice feature!

**Comment by matthewrdev** [comment]
This will be available in MFractor 4.3
